docs/concepts.rst

```diff
@@ -842,61 +842,124 @@ values by "case" instead of by entity.
+and so on.
+
+Bionic can take advantage of more computing resources by using local
+multiprocessing or Google AI Platform. The speedup depends on the structure of
+the :ref:`dependency graph <dagviz>` and/or the type of computation performed.
+See :ref:`Caveats <parallel-caveats>` for more details.
 
 .. _parallel-execution:
 
-Parallel Execution
-..................
+Local Parallel Execution
+........................
 
 .. versionadded:: 0.8.0
 
-Requesting an entity value with :meth:`Flow.get <bionic.Flow.get>` can lead to a long
-computation, as Bionic may need to compute that entity's dependencies, and their
-dependencies, and so on. By default, Bionic computes these values one at a time.
-However, it can also be configured to compute them in parallel; depending on the
-structure of your flow, this can be significantly faster.
-
-Parallel execution can be enabled like this:
+Bionic can use multiple CPU cores to compute entities in parallel. This feature
+is useful if you have many expensive operations which do not depend on each
+other.
 
 .. code-block:: python
 
     builder.set("core__parallel_execution__enabled", True)
 
 When parallel execution is enabled, Bionic starts up several worker processes
-[#workers]_, each of which can work on one value at a time. Of course, a worker can
-only start computing a value once all its dependencies are complete, so the number of
-processes that can be working at once depends on the :ref:`dependency graph
-<dagviz>`: if there aren't many branches in the graph, then most of the processes
-won't do much work. It does take extra time to set up the processes and move
-information between them, so parallel execution is not guaranteed to be faster
-overall. However, in general, if you have many expensive operations which don't
-depend on each other, enabling parallelism will improve performance.
-
-By default, Bionic will create one worker process for each CPU on your machine. This
-is usually a sensible number, but it can also be set directly:
+[#workers]_, each of which can work on one value at a time. By default, Bionic
+will create one worker process for each CPU on your machine. This is usually a
+sensible number, but it can also be set directly:
 
 .. code-block:: python
 
     builder.set("core__parallel_execution__worker_count", 8)
 
+.. [#workers] The pool of workers is managed by
+  `Loky <https://loky.readthedocs.io/en/stable/>`_,
+  which is built on Python's
+  `multiprocessing <https://docs.python.org/3.8/library/multiprocessing.html>`_ module.
+  The pool is global and reusable, so it should only need to be initialized once in
+  the lifetime of the main process.
+
+.. _google-aip:
+
+Google AI Platform Execution
+............................
+
+.. versionadded:: 0.9.0
+
+Bionic can use compute resources in the cloud by sending entities to
+`Google AI Platform (AIP) <https://cloud.google.com/ai-platform>`_ for
+computation. The GCP project needs to be specified.
```

Comment:
s/The GCP project needs to be specified./Enabling AI Platform requires a GCP project:/